### PR TITLE
sd_lavc: fix null dereference on error

### PR DESCRIPTION
**Pull request by na-na-hi**
If sub decoder does not exist, priv remains NULL.

Fixes: 1394e5a111d7917674d1c05e03aec3ebd9c27312

### CODE REVIEW

**Comment by kasper93** [comment]
Oh, good catch. But I would prefer simpler way. We don't have to allocate it, just guard deref in case of error.

```
--- a/sub/sd_lavc.c
+++ b/sub/sd_lavc.c
@@ -144,7 +144,8 @@ static int init(struct sd *sd)
  error:
     MP_FATAL(sd, "Could not open libavcodec subtitle decoder\n");
     avcodec_free_context(&ctx);
-    mp_free_av_packet(&priv->avpkt);
+    if (priv)
+        mp_free_av_packet(&priv->avpkt);
     talloc_free(priv);
     return -1;
 }
```

**Comment by kasper93** [comment]
> I don't think that's simpler.

Ok, make sure that `MP_FATAL(sd, "Could not open libavcodec subtitle decoder\n");` is not printed in this case.

> All other errors use goto error which handles everything properly, why can't these be the same?

Because it is not an error, it is probing step, same as the `switch (cid) {` at the beginning of the function. Also the allocation is not needed there yet, but the main point was to avoid log message.

> like how it should've been done in the first place

arguable. I don't mind either way tho, just makes sure the `MP_FATAL` is not printed when we stay in bitmap mode.

File: sub/sd_lavc.c [hunk 1]
```diff
@@ -105,24 +105,19 @@ static int init(struct sd *sd)
         int64_t format;
```
Comment (sfan5) [inline sub/sd_lavc.c:105]: there are two `goto error;` above here so correct error behavior for probing is incomplete

Comment (kasper93) [reply sub/sd_lavc.c:105]: You mean this one?
https://github.com/mpv-player/mpv/blob/d61d2946653a33d961ef029b6f73c3a20b26f9eb/sub/dec_sub.c#L174-L175 

I guess that works, indeed.

**Review by kasper93 — APPROVED** [review]

**Pull request merged**